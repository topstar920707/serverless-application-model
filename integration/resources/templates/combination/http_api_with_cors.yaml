
Globals:
  HttpApi:
    CorsConfiguration:
      AllowHeaders:
        - x-apigateway-header
      AllowMethods:
        - GET
      AllowOrigins:
        - https://foo.com
      ExposeHeaders:
        - x-amzn-header

Resources:
  HttpApiFunction:
    Type: AWS::Serverless::Function
    Properties:
      InlineCode: |
         exports.handler = async (event) => {
           return {
           statusCode: 200,
           body: JSON.stringify(event),
           headers: {}
           }
         }
      Handler: index.handler
      Runtime: nodejs12.x
      Events:
        ImplicitApi:
          Type: HttpApi
          Properties:
            Method: GET
            Path: /path
            TimeoutInMillis: 15000
            PayloadFormatVersion: "1.0"

Outputs:
  ApiUrl:
    Description: URL of your API endpoint
    Value:
      Fn::Sub: 'https://${ServerlessHttpApi}.execute-api.${AWS::Region}.${AWS::URLSuffix}/Prod/'
  ApiId:
    Description: Api id of ServerlessHttpApi
    Value:
      Ref: ServerlessHttpApi      
