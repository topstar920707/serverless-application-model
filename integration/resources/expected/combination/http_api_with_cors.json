[
  { "LogicalResourceId":"HttpApiFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"HttpApiFunctionRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"HttpApiFunctionImplicitApiPermission", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"ServerlessHttpApi", "ResourceType":"AWS::ApiGatewayV2::Api" },
  { "LogicalResourceId":"ServerlessHttpApiApiGatewayDefaultStage", "ResourceType":"AWS::ApiGatewayV2::Stage" }
]